pkg/connector/connector.go

```diff
@@ -333,13 +333,15 @@ func (c *EKSConnector) createConnectorRole(cluster ExternalCluster) (string, err
 		return "", errors.Wrap(err, "error creating IAM role")
 	}
 
-	if err := c.Provider.IAM().WaitUntilRoleExists(&iam.GetRoleInput{
+	waiter := iam.NewRoleExistsWaiter(c.Provider.IAM())
+	const maxWaitDuration = 5 * time.Minute
```

Comment:
This should probably live at the top with the other consts.

Reply:
My only concern with that is the `const` name will have to be much longer, indicating the operation it's defined for. So something like `getRoleMaxWaitDuration` or `checkRoleMaxWaitDuration`.